nolabs/refined/application/use_cases/diffdock/use_cases.py

```diff
@@ -0,0 +1,206 @@
+            for ligand_id in request.ligand_ids:
+                ligand = Ligand.objects.with_id(ligand_id)
+                if not ligand:
+                    raise NoLabsException(ErrorCodes.ligand_is_undefined)
+                ligands.append(ligand)
+
+            job.set_input(protein=protein,
+                          ligands=ligands,
+                          samples_per_complex=request.samples_per_complex)
+            job.save(cascade=True)
+
+            return map_job_to_response(job)
+        except Exception as e:
+            print(e)
+            if not isinstance(e, NoLabsException):
+                raise NoLabsException(ErrorCodes.unknown_exception) from e
+            raise e
+
+
+class GetJobStatusFeature:
+    """
+    Use case - set job status.
+    """
+    _diffdock = diffdock_microservice.DefaultApi
+
+    def __init__(self,
+                 diffdock: diffdock_microservice.DefaultApi):
+        self._diffdock = diffdock
+
+    async def handle(self, job_id: UUID) -> GetJobStatusResponse:
+        try:
+            job: DiffDockBindingJob = DiffDockBindingJob.objects.with_id(job_id)
+
+            if not job:
+                raise NoLabsException(ErrorCodes.job_not_found)
+
+            response = self._diffdock.is_job_running_job_job_id_is_running_get(
+                job_id=job.iid.value
+            )
+            return GetJobStatusResponse(
+                running=response.is_running
+            )
+        except Exception as e:
+            if not isinstance(e, NoLabsException):
+                raise NoLabsException(ErrorCodes.unknown_exception) from e
+            raise e
+
+
+class RunJobFeature:
+    """
+    Use case - start job.
+    """
+    _diffdock = diffdock_microservice.DefaultApi
+
+    def __init__(self, diffdock: diffdock_microservice.DefaultApi):
+        self._diffdock = diffdock
+
+    async def handle(self, job_id: UUID) -> JobResponse:
+        try:
+            assert job_id
+
+            job_id = JobId(job_id)
+            job: DiffDockBindingJob = DiffDockBindingJob.objects.with_id(job_id.value)
+
+            if not job:
+                raise NoLabsException(ErrorCodes.job_not_found)
+
+            result: List[DiffDockJobResult] = []
+
+            for ligand in job.ligands:
+                request = diffdock_microservice.RunDiffDockPredictionRequest(pdb_contents=job.protein.get_pdb(),
+                                                                             sdf_contents=ligand.get_sdf(),
+                                                                             samples_per_complex=job.samples_per_complex,
+                                                                             job_id=job_id.value,
+                                                                             )
+                response = self._diffdock.predict_run_docking_post(
+                    run_diff_dock_prediction_request=request, _request_timeout=(1000.0, 1000.0))
```

Comment:
Why do we set request timeout here? First Diffdock run on a fresh microservice could take up to 5 mins depending on data and GPU